core/src/main/scala/org/apache/spark/rdd/PipedRDD.scala

```diff
@@ -156,7 +157,34 @@ private[spark] class PipedRDD[T: ClassTag](
           out.close()
         }
       }
-    }.start()
+    }
+    stdinWriterThread.start()
+
+    def cleanUpIOThreads(): Unit = {
+      if (proc.isAlive) {
+        proc.destroy()
+      }
+      if (stdinWriterThread.isAlive) {
+        stdinWriterThread.stop()
```

Comment:
.stop() has been deprecated and 'dangerous' for a long time. You can interrupt the threads though

Reply:
Yes, I'm aware `Thread.stop` is deprecated and 'dangerous' . I didn't think it's sufficient to just interrupt threads and should be safe enough to stop these IO threads. 

After a second thought, interrupt should be enough. Will update the code